modules/core/shared/src/main/scala/weaver/internals/TagExpr.scala

```diff
@@ -0,0 +1,100 @@
+package weaver.internals
+
+import cats.parse.{ Parser0 => P0, Parser => P }
+import cats.syntax.all.*
+
+sealed trait TagExpr {
+  def eval(tags: Set[String]): Boolean
+}
+
+object TagExpr {
+
+  object Wildcard {
+    /* The complexity here comes from the fact that * at the beginning or in the middle needs to
+     * know the next expected parser to know when to stop consuming */
+    val parser = {
+      val validCharP = P.charIn('a' to 'z') |
+        P.charIn('A' to 'Z') |
+        P.charIn('0' to '9') |
+        P.charIn("_-:")
+
+      sealed trait Token
+      case class Literal(str: String) extends Token
+      case object Star                extends Token
+      case object Question            extends Token
+
+      val literalP: P[Token] =
+        validCharP.rep.map(cs => Literal(cs.toList.mkString))
+      val starP: P[Token]     = P.char('*').as(Star)
+      val questionP: P[Token] = P.char('?').as(Question)
+
+      val tokenP: P[Token]        = starP | questionP | literalP
+      val tokensP: P[List[Token]] = tokenP.rep.map(_.toList)
```

Comment:
Nitpick: I'd stick to `Parser` instead of `P`, would read more straightforward

Reply:
Done